src/librustdoc/clean/types.rs

```diff
@@ -57,20 +57,34 @@ pub(crate) type ItemIdSet = FxHashSet<ItemId>;
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Copy)]
 pub(crate) enum ItemId {
     /// A "normal" item that uses a [`DefId`] for identification.
-    DefId(DefId),
+    DefId(DefId, Option<Namespace>),
```

Comment:
Why is is necessary to change this? And if it is, could you please do that is a separate PR, as that would make it much easier to review.

Reply:
Without this disambiguation, rustdoc would start confusing things like a primitive & a module that documents its, a struct & a function with the same name, etc., it looked to me like a case of DefIds not carrying namespace info

Reply:
nevermind, I've realised later that the issues that led me to believe that DefIds don't carry namespace info were related to imports of different items with the same name, I've changed the impl now